### PR TITLE
Add loading component

### PR DESCRIPTION
**Pull request by clarityflowers**
The button component depends on this one, so I was getting issues using it in the new counts project.

Changes in the `dist` folder don't need to be reviewed since it's just auto-generated from the source. One advantage of using npm vs using github is that we wouldn't need to commit this folder if we used npm.

### CODE REVIEW

File: demo/src/App.tsx [hunk 1]
```diff
@@ -4,7 +4,6 @@ import { Button, Checkbox, Icon, Toggle } from "ride-ui-kit";
 
 import logo from "ride-ui-kit/dist/assets/logo.svg";
 import RRlogo from "ride-ui-kit/dist/assets/ride_report.png";
-import { TabToggle } from "ride-ui-kit";
```
Comment (clarityflowers) [inline demo/src/App.tsx:7]: Removed this from the demo because this component hasn't been added yet

**Review by devsweeting — APPROVED** [review]
One of the reasons I had left this out to circle back to was that when I imported and used the loading component in the demo react app it would cause nothing in the DOM tree to render. I was able to find that part of this issue was because of conflicting react packages. It seems like you can't have two separate react packages in a project and if two apps are using React they need to point to the same one. This could be a minor and exclusive bug to the demo app if it renders fine in the repo you're importing this into! Just wanted to point out this problem I had.

also I would have liked to use npm! - never too late! Linking this [thread on the conversation](https://ridereport.slack.com/archives/CCX734S6M/p1653422637690039) for reference

**Comment by clarityflowers** [comment]
This is specifically a bug with the counts repo!

It might be a good idea to move the demo app into its own repo just to avoid these issues.

**Comment by devsweeting** [comment]
I'm highly considering it now. never know if somethings bad idea until you try it

**Pull request merged**